### PR TITLE
[FIX] add constructors

### PR DESCRIPTION
**Pull request by h-2**
Who let me get away with making classes without constructors?

### CODE REVIEW

File: include/seqan3/range/view/deep.hpp [hunk 1]
```diff
@@ -111,6 +111,13 @@ class deep : public detail::adaptor_base<deep<underlying_adaptor_t>, underlying_
     /*!\name Constructors, destructor and assignment
      * \{
      */
+    constexpr deep()                         noexcept = default; //!< Defaulted.
```
Comment (smehringer) [inline include/seqan3/range/view/deep.hpp]: aren't those inherited by `using base_type::base_type` ?

Comment (h-2) [reply include/seqan3/range/view/deep.hpp]: Inherited copy constructors are not considered for copy construction of the derived type. Only for construction of the derived type from the base type. For the default constructors it is not as important, but we have rule-of-six so I added all of them.

**Comment by h-2** [comment]
BTW: jenkins as issues again (as predicted :zipper_mouth_face: )

File: include/seqan3/range/view/detail.hpp [hunk 2]
```diff
@@ -239,8 +240,19 @@ class combined_adaptor :
     }
 
 public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr combined_adaptor()                                              = default; //!< Defaulted.
```
Comment (smehringer) [inline include/seqan3/range/view/detail.hpp]: why is the default constructor not noexcept (also for the others)?

Comment (h-2) [reply include/seqan3/range/view/detail.hpp]: Because they aren't always `noexcept`. It depends on the base class.

Comment (h-2) [reply include/seqan3/range/view/detail.hpp]: And `noexcept` it not like `constexpr` in that it is ok if possible and if not also ok; it hard fails if the base type's constructor is not noexcept

**Pull request merged**